feature/session/src/main/java/io/github/droidkaigi/confsched2019/session/ui/widget/TimetableLayoutManager.kt

```diff
@@ -227,6 +258,49 @@ class TimeTableLayoutManager(
         return dx
     }
 
+    private fun layoutChildren(recycler: Recycler) {
+        val offsetY = parentTop
+        var offsetX = parentLeft
+        for (i in 0 until columns.size()) {
+            val columnNumber = columns.keyAt(i)
+            if (i == 0) anchor.leftColumn = columnNumber
+            offsetX += fillColumnHorizontally(columnNumber, 0, offsetX, offsetY, true, recycler)
+            if (offsetX > parentRight) {
+                anchor.rightColumn = columnNumber
+                break
+            }
+        }
+    }
+
+    private fun relayoutChildren(startX: Int, startY: Int, topPeriod: Period, recycler: Recycler) {
+        var offsetX = startX
+        anchor.leftColumn = topPeriod.columnNumber
+        val columnCount = columns.size()
+        val indexOfFirstColumn = columns.indexOfKey(topPeriod.columnNumber)
+        (indexOfFirstColumn until columnCount)
+            .plus(if (shouldLoopHorizontally && indexOfFirstColumn > 0) (0 until indexOfFirstColumn) else emptyList())
+            .forEach {
```

Comment:
how about this?

Reply:
The logic looks good. 💯

I guess `anchor.rightColumn` should be assigned if this loop finishes without through https://github.com/DroidKaigi/conference-app-2019/pull/687/files#diff-535b121f65cd5ca3a47e0df6b54a7ab4R301.
So it's better to define this range as a local variable once, maybe.

```
val range = (indexOfFirstColumn until columnCount)
            .plus(
                if (shouldLoopHorizontally && indexOfFirstColumn > 0) (0 until indexOfFirstColumn)
                else emptyList()
            )

range.forEach { ... }

anchor.rightColumn = requireNotNull(range.last())
```

Reply:
`Anchor.rightColumn` is not always the last of the range.
If `columnWidth` is 100px, the range is `[2, 3, 4, 5, 0, 1]` and the width of RecyclerView is 500px,  `Anchor.rightColumn` is `0`

Reply:
Yes, I know. Maybe you are saying about a case of https://github.com/DroidKaigi/conference-app-2019/pull/687/files#diff-535b121f65cd5ca3a47e0df6b54a7ab4R302
I meant a case when this loop finishes *without* through the branch.

Reply:
@MoyuruAizawa

Reply:
Ah, OK. _COMPLETELY UNDERSTOOD_
In case width of columns is less than width of RecyclerView, we cant't scroll. So, `Anchor.leftColumn` and `Anchor.rightColumn` is not referenced.
But I agree your suggestion. I fix it.